py3status/py3.py

```diff
@@ -679,3 +702,62 @@ def threshold_get_color(self, value, name=None):
         setattr(self._py3status_module, color_name, color)
 
         return color
+
+    def get_url(self, url):
+        """
+        Get the HTTP Response for the HTTP GET request to `url`.
+
+        Raises `Py3.URLError` on Connection Errors.
+        Raises `Py3.HttpClientError` on HTTP 4xx Errors.
+        Raises `Py3.HttpServerError` on HTTP 5xx Errors.
+        """
+        try:
+            response = urlopen(url)
+        except URLError as e:
+            self.log('Unable to access [{}]'.format(url),
+                     level=self.LOG_ERROR)
+            raise self.URLError(e)
+        return_code = response.getcode()
+        status_message = '{url} returned a {http_status} status.'.format(
+            url=url, http_status=return_code)
+        if 400 <= return_code < 500:
+            self.log(status_message, level=self.LOG_ERROR)
+            raise self.HttpClientError(status_message)
+        elif 500 <= return_code < 600:
+            self.log(status_message, level=self.LOG_WARNING)
+            raise self.HttpServerError(status_message)
+        return response
+
+    def get_json_data(self, url):
+        """
+        Deserialize JSON from `url` into a Python object.
+
+        `url` can be str or a HTTP Response object.
```

Comment:
I'd have get_url() return a simple object that has a json method but for the moment maybe just have get_url() private and worry about that later.  The module and most others that this would benefit only want get_json_data() anyhow.

That way we would be pretty much ready to go with this.